src/main/java/eu/deltasource/internship/hotel/service/BookingService.java

```diff
@@ -1,21 +1,199 @@
 package eu.deltasource.internship.hotel.service;
 
+import eu.deltasource.internship.hotel.domain.Booking;
+import eu.deltasource.internship.hotel.domain.Room;
 import eu.deltasource.internship.hotel.repository.BookingRepository;
 
+import java.time.LocalDate;
+import java.util.List;
+
 /**
- * Created by Taner Ilyazov - Delta Source Bulgaria on 2019-07-28.
+ * A class that represents service for bookings
  */
 public class BookingService {
 
-    private final BookingRepository bookingRepository;
+	private final BookingRepository bookingRepository;
+
+	private final RoomService roomService;
+
+	private final GuestService guestService;
+
+	/**
+	 * Constructs a booking service with given booking repository, room service and guest service
+	 *
+	 * @param bookingRepository the booking repository
+	 * @param roomService       the room service
+	 * @param guestService      the guest service
+	 */
+	public BookingService(BookingRepository bookingRepository, RoomService roomService, GuestService guestService) {
+		this.bookingRepository = bookingRepository;
+		this.roomService = roomService;
+		this.guestService = guestService;
+	}
+
+	/**
+	 * Creates a booking with given booking id, guest id, room id, number of people and dates
+	 *
+	 * @param bookingId      the booking id
+	 * @param guestId        the guest id
+	 * @param roomId         the room id
+	 * @param numberOfPeople the room's capacity
+	 * @param fromDate       the date of accommodation
+	 * @param toDate         the date of leaving
+	 */
+	public void createBooking(int bookingId, int guestId, int roomId, int numberOfPeople, LocalDate fromDate, LocalDate toDate) {
+		validateBookingId(bookingId);
+		if (existsById(bookingId)) {
+			throw new IllegalArgumentException("Booking with id " + bookingId + " already exists.");
+		}
+
+		validateGuestId(guestId);
+		validateRoomId(roomId);
```

Comment:
Let's think of a way to fetch the room, instead if getting it from the input request